Sources/SpeziScheduler/Model/TaskList.swift

```diff
@@ -0,0 +1,54 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Observation
+
+
+@Observable
+final class TaskList<Context: Codable> {
```

Comment:
It would be great to document a short reasoning why we introduce `TaskList` here and it would be great for me to understand its usage here and why it is used instead of a normal collection 👍

Reply:
There are two components to this answer and it ties back to your question above why we don't have the `Scheduler` observable.

* Having a dedicated Observable model, removes the need to make `Scheduler` observable. Which a) is a nice separation of concerns and b) avoids the hassle of using `@ObservationIgnored` everywhere fore the Module property wrappers.
* Having a dedicated Model class for this, allows us to easily pass around the reference to the `SchedulerStorage` class. This, e.g., enables the `SchedulerStorage/signalChange()` done by the individual `Event`'s.

Reply:
Thank you for the context!

Making Scheduler non-observable makes sense. I see that marking `Scheduler` as `@Observable` seem to bring is own challenges (e.g. the problem of not being able to use `@AppStorage`). 

I am not yet sure how this simplifies the `SchedulerStorage` reference as I don't see a challenge in doing this when using a normal collection and/or a dedicated `@Model` type that contains the tasks and a private reference to the storage basically making `Scheduler` only the configuration front of the module.

But the effect of the observation features itself makes enough sense to me that we can go with that. It would be interesting to see if taking this functionality into its own `@Modle` annotated class in the Scheduler might make even more sense and just injecting this in the SwiftUI environment but we can track this in an other PR.

Reply:
> I am not yet sure how this simplifies the SchedulerStorage reference as I don't see a challenge in doing this when using a normal collection and/or a dedicated @Model type that contains the tasks and a private reference to the storage basically making Scheduler only the configuration front of the module.

The basic thing is, that we can pass around the `SchedulerStorage` without additionally needing to pass the model instance.
We are currently not using Spezi's `@Model` right now as access is still done through the `Scheduler` module. We can definitely think about introducing a separate, accessible model for that. But I would move that to another PR, as you said 👍